### DISCUSSION TITLE
Having trouble understanding ESM and Remix.

### DISCUSSION THREAD

**Discussion by jscul** [remix-run/remix#6967] [Q&A]
![Screenshot from 2023-07-26 11-30-40](https://github.com/remix-run/remix/assets/22265367/eeb9ba50-2b95-4557-8ef6-ceee896b335b)

I'm getting the above warning with the "ESM" Express template and `react-markdown` as a dependency... however, my Remix app seems to work fine. ReactMarkdown is ESM only... but the Remix app is also ESM, right?

![image](https://github.com/remix-run/remix/assets/22265367/f5238bf5-a0e0-4ebc-ac75-58cf5c7345f0)

If I use `serverModuleFormat: "cjs",` with a different template... I can't use `react-markdown`.

Basically, my question is, can I ignore this warning? Is it incorrect?

**Comment by jscul** [comment]
```
import { getDependenciesToBundle } from "@remix-run/dev";

/** @type {import('@remix-run/dev').AppConfig} */
export default {
  ignoredRouteFiles: ["**/.*"],
  // appDirectory: "app",
  // assetsBuildDirectory: "public/build",
  // serverBuildPath: "build/index.js",
  // publicPath: "/build/",
  serverModuleFormat: "esm",
  future: {
    v2_dev: true,
    v2_errorBoundary: true,
    v2_headers: true,
    v2_meta: true,
    v2_normalizeFormMethod: true,
    v2_routeConvention: true,
  },
  serverDependenciesToBundle: [],
};
```

^ this is my remix.config.js.

**Comment by jscul** [comment]
https://github.com/remix-run/remix/discussions/6766#discussion-5368852

^ this resolves it... basically, fix is coming soon.
